sysapi/include/sysapi_util.h

```diff
@@ -103,9 +103,16 @@ typedef struct {
     size_t nextData;
 } _TSS2_SYS_CONTEXT_BLOB;
 
-#define SYS_CONTEXT ((_TSS2_SYS_CONTEXT_BLOB *)sysContext)
-#define SYS_RESP_HEADER ((TPM20_Header_Out *)(SYS_CONTEXT->cmdBuffer))
-#define SYS_REQ_HEADER ((TPM20_Header_In *)(SYS_CONTEXT->cmdBuffer))
+struct TSS2_SYS_CONTEXT;
+
+static inline _TSS2_SYS_CONTEXT_BLOB *
+syscontext_cast(TSS2_SYS_CONTEXT *ctx)
+{
+    return (_TSS2_SYS_CONTEXT_BLOB*) ctx;
+}
+
+#define SYS_RESP_HEADER(ctx) ((TPM20_Header_Out *)(ctx->cmdBuffer))
+#define SYS_REQ_HEADER(ctx) ((TPM20_Header_In *)(ctx->cmdBuffer))
```

Comment:
Why not make these static inline? Same performance as a macro *with* type checking.

Reply:
You are right. These should be static inlines as well. I think I just focused on SYS_CONTEXT too much.